hikari/impl/rest.py

```diff
@@ -258,7 +258,7 @@ def proxy_settings(self) -> config.ProxySettings:
         return self._rest().proxy_settings
 
 
-_NONE_OR_UNDEFINED: typing.Final[typing.Set[undefined.UndefinedOr[None]]] = {None, undefined.UNDEFINED}
+_NONE_OR_UNDEFINED: typing.Final[typing.Tuple[None, undefined.UndefinedType]] = (None, undefined.UNDEFINED)
```

Comment:
The reason this was a set initially is to be able to have fast checking, as that is what sets are used for. Not sure if tuple is ok will leave it up to @FasterSpeeding :P

Reply:
The set requires the comparing item to be hashable, whereas the tuple doesn't. According to my IDE, this definition is unused other than the 3 usages here, which require a tuple rather than a set for hashability

Reply:
Looks fine to me